workers/unity/Assets/Playground/Scripts/Cubes/CollisionProcessSystem.cs

```diff
@@ -28,49 +34,51 @@ private struct Data
             [ReadOnly] public ComponentDataArray<Authoritative<SpatialOSLaunchable>> DenotesAuthority;
         }
 
         [Inject] private Data data;
 
-        private static EntityId EmptyEntityId = new EntityId(0);
-
         protected override void OnUpdate()
         {
             for (var i = 0; i < data.Length; i++)
             {
                 // Handle all the different possible outcomes of the collision.
                 // This requires looking at their most recent launchers.
                 var collision = data.Collision[i];
-                var first = collision.OwnEntity;
-                var second = collision.OtherEntity;
-                var firstLaunchable = EntityManager.GetComponentData<SpatialOSLaunchable>(first);
-                var secondLaunchable = EntityManager.GetComponentData<SpatialOSLaunchable>(second);
+                var sender = data.Sender[i];
+
+                var firstLaunchable = EntityManager.GetComponentData<SpatialOSLaunchable>(collision.OwnEntity);
+                var secondLaunchable = EntityManager.GetComponentData<SpatialOSLaunchable>(collision.OtherEntity);
                 var firstLauncher = firstLaunchable.MostRecentLauncher;
                 var secondLauncher = secondLaunchable.MostRecentLauncher;
+
                 if (firstLauncher == secondLauncher)
                 {
-                    if (firstLauncher != EmptyEntityId)
+                    if (firstLauncher.IsValid())
                     {
-                        data.Sender[i].RequestsToSend.Add(new Launcher.IncreaseScore.Request(
-                            firstLauncher, new Generated.Playground.ScoreIncreaseRequest { Amount = 1 }));
+                        sender.RequestsToSend.Add(new Launcher.IncreaseScore.Request(
+                            firstLauncher, new ScoreIncreaseRequest { Amount = 1 }));
+                        data.Sender[i] = sender;
                     }
                 }
-                else if (secondLauncher != EmptyEntityId)
+                else if (secondLauncher.IsValid())
                 {
                     var launchable = data.Launchable[i];
-                    if (firstLauncher == EmptyEntityId)
+                    if (!firstLauncher.IsValid())
                     {
-                        launchable.MostRecentLauncher = secondLauncher;
-                        data.Sender[i].RequestsToSend.Add(new Launcher.IncreaseScore.Request(secondLauncher,
+                        sender.RequestsToSend.Add(new Launcher.IncreaseScore.Request(secondLauncher,
                             new ScoreIncreaseRequest { Amount = 1 }));
+                        data.Sender[i] = sender;
+                        
+                        launchable.MostRecentLauncher = secondLauncher;
                     }
                     else
                     {
-                        launchable.MostRecentLauncher = EmptyEntityId;
+                        launchable.MostRecentLauncher = new EntityId(0);
```

Comment:
I'd keep `EmptyEntityId` here

Reply:
How so? It is the only place it is used, and there is no allocation since its a struct

Reply:
Because it looks a bit wrong to float a magic number around.

Reply:
I'll add a `private const EntityId InvalidEntityId = new EntityId(0);`

Reply:
+1 for the `InvalidEntityId`